oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -263,6 +296,30 @@ pub fn readers_statuses(readers: &[Option<&ChannelReader>]) -> Vec<Result<bool,
         .collect()
 }
 
+/// Block until the runtime is terminated, or the reader is orphaned or has a message.
+pub fn block_thread_on_channel(
+    runtime: &RuntimeRef,
+    reader: &ChannelReader,
+) -> Result<(), OakStatus> {
+    if runtime.is_terminating() {
+        return Ok(());
+    }
+
+    let thread = platform::thread::current();
+    let thread_id = platform::thread::current().id();
+    let thread_ref = Arc::new(thread);
+
+    reader.add_waiter(thread_id, &thread_ref);
+
+    if reader.has_message()? {
+        return Ok(());
+    }
+
```

Comment:
If another thread writes to the channel at this point (after the `has_message` check, before the `park`), will the thread get blocked? 

(From the [park docs](https://doc.rust-lang.org/std/thread/fn.park.html) I *think* this is OK, but maybe explicitly call it out in a comment.)

Reply:
Added a comment. Since we have already added a waiter, `unpark` will have been called, and so our `park` will return immediately.